apps/frontend/src/modules/events/components/index.tsx

```diff
@@ -0,0 +1,226 @@
+import { EventDefinitionCountableEdge, useEventsLazyQuery } from "@/generated/graphql";
+import { Properties } from "@/types";
+import { Dialog, DialogContent, DialogTrigger, Header, Spinner } from "@/ui";
+import { QuestionIcon } from "@/ui/icons";
+import { CursorIconLg } from "@/ui/icons/CursorIconLg";
+import { cn } from "@/utils";
+import {
+    Icon,
+    Table,
+    TableBody,
+    TableCell,
+    TableFoot,
+    TableFooterCell,
+    TableHead,
+    TableHeaderCell,
+    TableRow,
+} from "@tremor/react";
+import { format } from "date-fns";
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import { useState } from "react";
+import { useEvents } from "../hooks/useEvents";
+
+export const EventsIndex = () => {
+    const [selectedEvent, setSelectedEvent] = useState<EventDefinitionCountableEdge | null>(null);
+
+    const {
+        events,
+        eventDefinitions,
+        getPropertiesWithDefinitions,
+        isFetchingMore,
+        propertiesWithDefinitions,
+        loadingProperties,
+        loadingEventDefinitions,
+        eventDefinitionsOnPage,
+        getMoreEventDefinitions,
+    } = useEvents();
+
+    const [page, setPage] = useState<number>(1);
```

Comment:
@afolasope can we use zustand for this state so it can be persisted across navigations, currently if I'm on page 2 and I navigate to another page, lets say survey list page, when I come back to this page, and the component mounts again, `page` would be `1` instead of `2`. 

When this happens, the details at the bottom of the table would say "1 - 8 of <n> events" instead of "9 - 16 of <n> events"